etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequests.java

```diff
@@ -0,0 +1,48 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.DiagnosticReport;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.ServiceRequest;
+
+/**
+ * This transformation removes all OBRs from an ORU message except for the OBR with value "54089-8"
+ * in OBR-4.1. All OBXs are attached to the sole remaining OBR
+ */
+public class RemoveObservationRequests implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+
+        Stream<DiagnosticReport> diagnosticReports =
+                HapiHelper.resourcesInBundle(bundle, DiagnosticReport.class);
+        List<Observation> observations =
+                HapiHelper.resourcesInBundle(bundle, Observation.class).toList();
+
+        for (DiagnosticReport report : diagnosticReports.toList()) {
+            ServiceRequest serviceRequest = HapiHelper.getServiceRequest(report);
+            if (serviceRequest != null) {
+                String ob4_1 = HapiHelper.getOBR4_1Value(serviceRequest);
+                if ("54089-8".equals(ob4_1)) {
```

Comment:
Thoughts on using a variable that gets passed as a parameter? This will generalize the transformation in a way that it can be used for other LOINC codes and it will remove the use of hard coded values.

Reply:
That's a good idea. I can make that change

Reply:
Updated it to add the argument